lib/fluent/plugin/base.rb

```diff
@@ -52,15 +52,14 @@ def fluentd_worker_id
         @_fluentd_worker_id
       end
 
+      # @param conf [Fluent::Config::Element] configuration
       def configure(conf)
-        if Fluent::Engine.supervisor_mode || (conf.respond_to?(:for_this_worker?) && conf.for_this_worker?)
-          workers = if conf.target_worker_ids && !conf.target_worker_ids.empty?
-                      conf.target_worker_ids.size
-                    else
-                      1
-                    end
-          system_config_override(workers: workers)
+        raise ArgumentError, 'type of conf is Fluent::Config::Element' unless conf.is_a?(Fluent::Config::Element)
```

Comment:
```suggestion
        raise ArgumentError, 'BUG: type of conf must be Fluent::Config::Element, but #{conf.class} is passed.' unless conf.is_a?(Fluent::Config::Element)
```

From the other code, might a message like this be better?

https://github.com/fluent/fluentd/blob/fb641737d6bacace19be11cb676fec3fc20f0cf9/lib/fluent/engine.rb#L123-L125

Reply:
I see! I changed it that way.